projects/com.oracle.truffle.llvm.tools/src/com/oracle/truffle/llvm/tools/LLVMTools.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of
+ * conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of
+ * conditions and the following disclaimer in the documentation and/or other materials provided
+ * with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ * endorse or promote products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.truffle.llvm.tools;
+
+import java.io.File;
+
+import com.oracle.truffle.llvm.runtime.options.LLVMBaseOptionFacade;
+import com.oracle.truffle.llvm.tools.util.PathUtil;
+import com.oracle.truffle.llvm.tools.util.ProcessUtil;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class LLVMTools {
+
+    private static final File PROJECT_ROOT = new File(LLVMBaseOptionFacade.getProjectRoot() + File.separator + LLVMTools.class.getPackage().getName());
+    private static final File TOOLS_ROOT = new File(PROJECT_ROOT, "tools");
+    private static final File LLVM_ROOT = new File(TOOLS_ROOT, "llvm/bin");
+    private static final String[] LLVM_SUPPORTED_VERSIONS = {"3.2"};
+
+    public enum Tool {
+        ASSEMBLER("llvm-as"),
+        COMPILER("llc"),
+        CLANG("clang"),
+        CLANG_PP("clang++"),
+        OPT("opt");
+
+        private final Path toolPath;
+
+        Tool(String programName) {
+            this.toolPath = findTool(programName);
+        }
+
+        private static Path findTool(String toolName) {
+            Path path = ToolLookup.findTool(toolName + "(-mp.*)?", s -> Arrays.stream(LLVM_SUPPORTED_VERSIONS).filter(v -> s.contains(v)).findAny().isPresent());
```

Comment:
Previously we looked for clang executables which were named `clang`, `clang32`, and `clang3.2`. Is this logic still there (this also applies to GCC)?